### PR TITLE
Make ApiRequestContext provide KyuubiRestFrontendService directly

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Make full use of KyuubiRestFrontendService without hacking, e.g. for the swagger UI base.

We can do further improvement based on this PR.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @yanghua @cfmcgrady @simon824

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala [hunk 1]
```diff
@@ -63,7 +63,8 @@ private[v1] class SessionsResource extends ApiRequestContext {
   @Path("{sessionHandle}")
   def sessionInfo(@PathParam("sessionHandle") sessionHandleStr: String): KyuubiSessionEvent = {
     try {
-      KyuubiSessionEvent(backendService.sessionManager.getSession(
+      // TODO: need to use KyuubiSessionEvent in session
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala:66]: this is a bug, the KyuubiSessionEvent.apply will lose live info of a session cc @simon824

Comment (simon824) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala:66]: Got it, i will fix it later.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/api.scala [hunk 2]
```diff
@@ -36,42 +33,18 @@ private[api] trait ApiRequestContext {
...
+  final protected def fe: KyuubiRestFrontendService = FrontendServiceContext.get(servletContext)
 }
 
-private[api] object BackendServiceProvider {
+private[api] object FrontendServiceContext {
 
   private val attribute = getClass.getCanonicalName
 
-  def setBackendService(contextHandler: ContextHandler, be: BackendService): Unit = {
+  def set(contextHandler: ContextHandler, be: KyuubiRestFrontendService): Unit = {
```
Comment (yanghua) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/api.scala:43]: be -> fe?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/ApiRootResource.scala [hunk 3]
```diff
@@ -53,14 +56,32 @@ private[v1] class ApiRootResource extends ApiRequestContext {
...
   @Produces(Array(MediaType.TEXT_HTML))
   def swaggerUi(): Response = {
-    val restServiceOpt = KyuubiServer.kyuubiServer
-      .frontendServices
-      .collectFirst { case rest: KyuubiRestFrontendService => rest }
-    assert(restServiceOpt.isDefined)
-    val serverIP = restServiceOpt.map(_.connectionUrl).get
-    val swaggerUi =
-      s"http://$serverIP/swagger-ui-redirected/index.html?url=http://$serverIP/openapi.json"
+    val swaggerUi = s"http://${fe.connectionUrl}/swagger-ui-redirected/index.html?url=" +
```
Comment (yanghua) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/ApiRootResource.scala:59]: swaggerUi -> swaggerUI?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/ApiRootResource.scala [hunk 4]
```diff
@@ -53,14 +56,32 @@ private[v1] class ApiRootResource extends ApiRequestContext {
...
+    FrontendServiceContext.set(handler, fe)
+    handler.addServlet(servlet, "/*")
+
+    // install swagger-ui, these static files are copied from
+    // https://github.com/swagger-api/swagger-ui/tree/master/dist
+    val swaggerUI = new ServletHolder("swagger-ui", classOf[DefaultServlet])
+    swaggerUI.setInitParameter(
+      "resourceBase",
+      getClass.getClassLoader()
+        .getResource("META-INF/resources/webjars/swagger-ui/4.1.3/")
```
Comment (yanghua) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/ApiRootResource.scala:81]: I suggest we do not hardcode the detailed version in the future.

**Review by yanghua — COMMENTED** [review]
one more minor comment, overall LGTM.

**Review by cfmcgrady — APPROVED** [review]

**Review by yanghua — APPROVED** [review]
lgtm

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**